compiler/compiler.go

```diff
@@ -75,6 +75,12 @@ func Compile(scope Scope, f *semantic.FunctionExpression, in semantic.MonoType)
 // If the input type is not a type variable, it will check to ensure
 // that the type in the input matches or it will return an error.
 func substituteTypes(subst *semantic.Substitution, inferredType, actualType semantic.MonoType) error {
+	// FIXME: hack to allow `[dynamic]` to flow through `array.map`.
```

Comment:
n.b. I think the original problem with `array.map` was resolved by rolling back some bad edits.

iirc _some sort of check like this is still necessary_, however.